matchzoo/models/esim.py

```diff
@@ -0,0 +1,218 @@
+        projected_right = projection(enhanced_right)
+        projected_left = keras.layers.Dropout(
+            rate=self._params['dropout_rate'])(projected_left)
+        projected_right = keras.layers.Dropout(
+            rate=self._params['dropout_rate'])(projected_right)
+        # Fusion the enhanced features by another BiLSTM
+        fusion_layer = keras.layers.Bidirectional(keras.layers.LSTM(
+            hidden_size,
+            return_sequences=True,
+            dropout=self._params['dropout_rate']
+        ))
+        # shape = [B, L, 2*H]
+        fusion_left = fusion_layer(projected_left)
+        # shape = [B, R, 2*H]
+        fusion_right = fusion_layer(projected_right)
+        # Do global avg and max pooling and get a fixed-length representation vector
+        avg_left_rep = keras.layers.Lambda(lambda x: K.sum(x * expand_tensor(left_mask), axis=1) /
+                                                     expand_tensor(K.sum(left_mask, axis=1)))(fusion_left)
+        avg_right_rep = keras.layers.Lambda(lambda x: K.sum(x * expand_tensor(right_mask), axis=1) /
+                                                      expand_tensor(K.sum(right_mask, axis=1)))(fusion_right)
+        max_left_rep = keras.layers.Lambda(lambda x: K.max(x + (1 - tile_tensor(left_mask)) * -1e7, axis=1))(
+            fusion_left)
+        max_right_rep = keras.layers.Lambda(lambda x: K.max(x + (1 - tile_tensor(right_mask)) * -1e7, axis=1))(
+            fusion_right)
+        # shape = [B, 8*H]
+        cls_input = keras.layers.Concatenate()([avg_left_rep, avg_right_rep, max_left_rep, max_right_rep])
+        cls_input = keras.layers.Dropout(
+            rate=self._params['dropout_rate'])(cls_input)
+        # Output layer
+        mlp = self._make_multi_layer_perceptron_layer()(cls_input)
+        mlp = keras.layers.Dropout(
+            rate=self._params['dropout_rate'])(mlp)
+        inputs = [input_left, input_right]
+        # mlp = keras.layers.Lambda(lambda x: K.tf.Print(x, [x]))(mlp)
+        x_out = self._make_output_layer()(mlp)
+        self._backend = keras.Model(inputs=inputs, outputs=x_out)
+
+    def _attention(self, encoded_left,
+                   left_mask,
+                   encoded_right,
+                   right_mask):
+        # Interaction
+        embed_cross = keras.layers.Dot(axes=2)([encoded_left, encoded_right])
+        cross_shape = [self._params['input_shapes'][0][0], self._params['input_shapes'][1][0]]
+        embed_cross = keras.layers.Reshape(cross_shape)(embed_cross)
+        left2right_attn = self._masked_softmax(embed_cross, left_mask,
+                                               self._params['input_shapes'][0][0],
+                                               self._params['input_shapes'][1][0])
+        # left2right_attn = keras.layers.Lambda(lambda x: K.tf.Print(x, [x]))(left2right_attn)
+
+        right2left_attn = self._masked_softmax(keras.layers.Permute([2, 1])(embed_cross), right_mask,

```

Comment:
I don't like the way you naming your variables, specifically, digits together with letters.